packages/modules/index/integration-tests/__tests__/query-builder.spec.ts

```diff
@@ -296,7 +297,24 @@ describe("IndexModuleService query", function () {
       },
     })
 
+    const { data: dataNot } = await module.query({
+      fields: ["product.*", "product.variants.*", "product.variants.prices.*"],
+      filters: {
+        product: {
+          variants: {
+            $not: [
+              {
+                sku: { $eq: null },
+              },
+            ],
```

Comment:
Suggestion: shouldn't `$not` just be an object such as `$not: { $eq: null }` and if you want multiple conditions then you wrap it into a `$and` just like we have with the `OperatorMap` mentionned above? That way we can kind of align the API's as we are in there as it will be a task anyway. wdy?

Reply:
I will keep the type as object, but I will let the query builder handle both. I think it makes sense for logical operators of any king to be grouped.